drivers/input/input_stmpe811.c

```diff
@@ -0,0 +1,457 @@
+	if (err < 0) {
+		return err;
+	}
+
+	if (int_sta & STMPE811_INT_BIT_FIFO_THRESHOLD) {
+		/**
+		 * Report every element in FIFO
+		 *
+		 * This requires a while loop to avoid a race condition
+		 * in which an element is added after reading FIFO_SIZE.
+		 *
+		 * Exiting ISR without handling every element in FIFO
+		 * would prevent FIFO_THRESHOLD interrupt from being triggered again.
+		 */
+		while (true) {
+			err = i2c_reg_read_byte_dt(&config->bus, STMPE811_FIFO_SIZE_REG,
+						   &fifo_size);
+			if (err < 0) {
+				return err;
+			}
+
+			if (fifo_size == 0) {
+				break;
+			}
+
+			for (int i = 0; i < fifo_size; i++) {
+				err = stmpe811_ts_get_data(dev);
+				if (err < 0) {
+					return err;
+				}
+
+				stmpe811_report_touch(dev);
+			}
+		}
+	}
+
+	/* TOUCH interrupt also gets triggered at release */
+	if (int_sta & STMPE811_INT_BIT_TOUCH) {
+		err = i2c_reg_read_byte_dt(&config->bus, STMPE811_TSC_CTRL_REG, &tsc_ctrl);
+		if (err < 0) {
+			return err;
+		}
+
+		/* TOUCH interrupt + no touch detected in TSC_CTRL reg <=> release */
+		if (!(tsc_ctrl & STMPE811_TSC_CTRL_BIT_TOUCH_DET)) {
+			input_report_key(dev, INPUT_BTN_TOUCH, 0, true, K_FOREVER);
+		}
+	}
+
+	return 0;
+}
+
+static void stmpe811_work_handler(struct k_work *work)
+{
+	struct stmpe811_data *data = CONTAINER_OF(work, struct stmpe811_data, processing_work);
+	const struct stmpe811_config *config = data->dev->config;
+
+	stmpe811_process(data->dev);
+	/**
+	 * Reschedule ISR if there was an interrupt triggered during handling (race condition).
+	 * IRQ is edge-triggered, so otherwise it would never be triggered again.
+	 */
+	if (gpio_pin_get_dt(&config->int_gpio)) {
+		k_work_submit(&data->processing_work);
+	}
+}
+
+static void stmpe811_interrupt_handler(const struct device *dev, struct gpio_callback *cb,
+				       uint32_t pins)
+{
+	struct stmpe811_data *data = CONTAINER_OF(cb, struct stmpe811_data, int_gpio_cb);
+
+	k_work_submit(&data->processing_work);
+}
+
+static int stmpe811_verify_chip_id(const struct device *dev)
+{
+	const struct stmpe811_config *config = dev->config;
+	uint8_t buf[2];
+
+	i2c_burst_read_dt(&config->bus, STMPE811_CHP_ID_LSB_REG, buf, 2);
+
+	if (sys_get_be16(buf) != CHIP_ID) {
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int stmpe811_init(const struct device *dev)
+{
+	const struct stmpe811_config *config = dev->config;
+	struct stmpe811_data *data = dev->data;
+	int err;
+
+	if (!i2c_is_ready_dt(&config->bus)) {
+		LOG_ERR("I2C controller device not ready");
+		return -ENODEV;
+	}
+
+	data->dev = dev;
+
+	k_work_init(&data->processing_work, stmpe811_work_handler);
+
+	/* Check configs */
+	if (config->raw_x_max <= config->raw_x_min) {
+		LOG_ERR("Invalid config: raw-x-max should be larger than raw-x-min.");
+		return -EINVAL;
+	}
+	if (config->raw_y_max <= config->raw_y_min) {
+		LOG_ERR("Invalid config: raw-y-max should be larger than raw-y-min.");
+		return -EINVAL;
+	}
```

Comment:
You are better off writing these as `BUILD_ASSERT`, that way if the condition happens it'll faill the build and you won't end up with a firwmare that does not work. Also saves few bytes of flash. :-) See this as an example: https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/input/input_longpress.c#L112-L115

Reply:
Changed to `BUILD_ASSERT`.